mock/mock_repository/mock_fetcher.go

```diff
@@ -0,0 +1,62 @@
+// Automatically generated by MockGen. DO NOT EDIT!
```

Comment:
Would prefer to avoid using auto generated mocks in the codebase. it encourages this approach to testing when we really need to use integration tests.